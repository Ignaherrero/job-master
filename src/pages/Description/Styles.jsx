import styled from "styled-components";

const Title = styled.h2`
  color: #b9bdcf;
`;

const Job = styled(Title)`
  color: #334680;
`;

const DescriptionJob = styled.p`
  color: #334680;
`;

const CardTwo = styled.div`
  display: flex;
  margin-top: 12px;

  & img {
    display: block;
    width: 80px;
  }
`;

const Contact = styled.p`
  color: #334680;
  font-weight: 500;
`;

export { Contact, CardTwo, DescriptionJob, Job, Title };
