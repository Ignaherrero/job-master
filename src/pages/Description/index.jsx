import React from "react";

import { useHistory } from "react-router-dom";

import Button from "../../components/Button";
import Span from "../../components/Span/Index";
import { CardTwo, Contact, DescriptionJob, Job, Title } from "./Styles.jsx";
export const Description = () => {
  const history = useHistory();
  const { location } = useHistory();
  const { companyLogo, title, type, company, description } = location.state;
  const handleClick = () => {
    history.push("/");
  };

  return (
    <>
      <Button onClick={handleClick}>Back to search</Button>
      <Title>{title}</Title>
      <Contact>
        Please email a copy of your resume and online portfolio to{" "}
        <a href="">wes@kasisto.com</a> & CC <a href="">eric@kasisto.com</a>
      </Contact>
      <Job></Job>
      <Button>{type}</Button>
      <Span clock block>
        5 days ago
      </Span>
      <CardTwo>
        <img src={companyLogo} alt={title} />
        <div>
          <h3>{company}</h3>
          <Span world>New York</Span>
        </div>
      </CardTwo>
      <div dangerouslySetInnerHTML={{ __html: description }}></div>
      <DescriptionJob>{description}</DescriptionJob>
    </>
  );
};
