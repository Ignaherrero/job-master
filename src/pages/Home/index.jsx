/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
/* eslint-disable no-debugger */
import React, { useState, useReducer } from "react";

import { Pagination } from "react-custom-pagination";
import { useHistory } from "react-router-dom";

import { ReactComponent as IconCheckOk } from "../../assets/check.svg";
import { SEARCH_JOB } from "../../common/constants";
import Card from "../../components/Card/Index";
import Footer from "../../components/Footer";
import Form from "../../components/Form/Index";
import Input from "../../components/Input/Index";
import { useFetch } from "../../hooks/useFetch";
import { jobReducer } from "../../reducers/jobReducers";
import { Container, TitleStyle } from "./Styles.jsx";

const initialState = {
  cityNameRadio: "london",
  cityNameInput: "",
  jobName: "",
  fullTime: true,
};

export const Home = () => {
  const [form, dispatch] = useReducer(jobReducer, initialState);

  const { data: jobs, isLoading } = useFetch(
    form.jobName,
    form.cityNameInput,
    form.fullTime
  );

  const handleChange = (e) => {
    const action = {
      type: SEARCH_JOB,
      payload: { name: e.target.name, value: e.target.value },
    };

    dispatch(action);
  };

  const history = useHistory();

  const [currentPage, setCurrentPage] = useState(1);
  const [postsPerPage] = useState(5);

  // get current Posts
  const indexOfLastPost = currentPage * postsPerPage;
  const indexOfFirstPost = indexOfLastPost - postsPerPage;
  const currentPosts = jobs.slice(indexOfFirstPost, indexOfLastPost);

  // when user clicks on number this function will execute
  const paginate = (number) => {
    setCurrentPage(number);
  };

  debugger;
  return (
    <>
      <Container>
        <TitleStyle>
          Github <span>Jobs</span>
        </TitleStyle>
        {JSON.stringify(form)}
        <Form>
          <div>
            <div>
              <div>
                <label htmlFor="">
                  <Input
                    type="text"
                    placeholder="title, company, expertice"
                    handleChange={handleChange}
                    name="jobName"
                  />
                </label>
                <Input type="submit" value="Search" />
              </div>
            </div>
          </div>

          <label onChange={handleChange}>
            <span>{form.fullTime && <IconCheckOk />}</span>
            Full Time
            <Input
              type="checkbox"
              name="fullTime"
              value={form.fullTime}
              dontShowPromptCheckbox
            />
          </label>

          <span>Location</span>
          <label htmlFor="" onChange={handleChange}>
            <Input
              name="cityNameInput"
              type="text"
              placeholder="City, state, zip code or country"
            />
          </label>

          <fieldset>
            <label htmlFor="london" onChange={handleChange}>
              <Input
                type="radio"
                id="london"
                value="london"
                name="cityNameInput"
                checked={form.cityNameRadio === "london"}
              />
              London
            </label>
            <label htmlFor="amsterdam" onChange={handleChange}>
              <Input
                type="radio"
                id="amsterdam"
                value="amsterdam"
                name="cityNameInput"
                checked={form.cityNameRadio === "amsterdam"}
              />
              Amsterdam
            </label>
            <label htmlFor="newyork" onChange={handleChange}>
              <Input
                type="radio"
                id="new york"
                value="new york"
                name="cityNameInput"
                checked={form.cityNameRadio === "newyork"}
              />
              New York
            </label>
            <label htmlFor="berlin" onChange={handleChange}>
              <Input
                type="radio"
                id="berlin"
                value="berlin"
                name="cityNameInput"
                checked={form.cityNameRadio === "berlin"}
              />
              Berlin
            </label>
          </fieldset>
        </Form>

        {/* <Formi
          handleSubmit={handleSubmit}
          handleChange={handleChange}
          form={form}
        /> */}

        {!isLoading && jobs.length === 0 && <h1>Ningun job en tu ubicaci√≥n</h1>}

        {currentPosts.map((item) => (
          <Card
            key={item.id}
            companyLogo={item.company_logo}
            title={item.title}
            type={item.type}
            location={item.location}
            createdAt={item.created_at}
            company={item.company}
            description={item.description}
            history={history}
          />
        ))}

        {Boolean(jobs.length) && (
          <Pagination
            totalPosts={jobs.length}
            postsPerPage={postsPerPage}
            paginate={paginate}
            view={5}
            showLast={true}
            showFirst={true}
            showIndex={true}
          />
        )}

        <Footer />
      </Container>
    </>
  );
};
