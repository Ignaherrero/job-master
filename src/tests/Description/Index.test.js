import { shallow } from "enzyme/build";
import { useHistory } from "react-router";

import { Description } from "../../pages/Description";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: {
      state: {
        companyLogo: "sasa",
        title: "algo",
        type: "full",
        company: "algo",
        description: "algo",
      },
    },
  }),
}));

describe("<Description/>", () => {
  test("debe retornar el componente description", () => {
    const wrapper = shallow(<Description />);

    expect(wrapper).toMatchSnapshot();
  });
});
