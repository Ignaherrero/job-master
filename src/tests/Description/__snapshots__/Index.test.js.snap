// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description/> debe retornar el componente description 1`] = `
<Fragment>
  <Button
    onClick={[Function]}
  >
    Back to search
  </Button>
  <styled.h2>
    algo
  </styled.h2>
  <styled.p>
    Please email a copy of your resume and online portfolio to
     
    <a
      href=""
    >
      wes@kasisto.com
    </a>
     & CC 
    <a
      href=""
    >
      eric@kasisto.com
    </a>
  </styled.p>
  <Styled(styled.h2) />
  <Button>
    full
  </Button>
  <Span
    block={true}
    clock={true}
  >
    5 days ago
  </Span>
  <styled.div>
    <img
      alt="algo"
      src="sasa"
    />
    <div>
      <h3>
        algo
      </h3>
      <Span
        world={true}
      >
        New York
      </Span>
    </div>
  </styled.div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "algo",
      }
    }
  />
  <styled.p>
    algo
  </styled.p>
</Fragment>
`;
